core/pfe-core/controllers/roving-tabindex-controller.ts

```diff
@@ -47,74 +52,131 @@ export class RovingTabindexController<
   /**
    * active item of array of items
    */
-  get activeItem(): ItemType | undefined {
+  get activeItem(): HTMLElement | undefined {
     return this.#activeItem;
   }
 
+  /**
+   * all items from array
+   */
+  get items() {
+    return this.#items;
+  }
+
+  /**
+   * all focusable items from array
+   */
+  get focusableItems() {
+    return this.#focusableItems;
+  }
+
   /**
    * first item in array of focusable items
    */
-  get firstItem(): ItemType | undefined {
+  get firstItem(): HTMLElement | undefined {
     return this.#focusableItems[0];
   }
 
   /**
    * last item in array of focusable items
    */
-  get lastItem(): ItemType | undefined {
+  get lastItem(): HTMLElement | undefined {
     return this.#focusableItems.at(-1);
   }
 
   /**
    * next item  after active item in array of focusable items
    */
-  get nextItem(): ItemType | undefined {
+  get nextItem(): HTMLElement | undefined {
     return (
-        this.#activeIndex >= this.#focusableItems.length - 1 ? this.firstItem
-      : this.#focusableItems[this.#activeIndex + 1]
+      this.#activeIndex >= this.#focusableItems.length - 1 ? this.firstItem
+        : this.#focusableItems[this.#activeIndex + 1]
     );
   }
 
   /**
    * previous item  after active item in array of focusable items
    */
-  get prevItem(): ItemType | undefined {
+  get prevItem(): HTMLElement | undefined {
     return (
-        this.#activeIndex > 0 ? this.#focusableItems[this.#activeIndex - 1]
-      : this.lastItem
+      this.#activeIndex > 0 ? this.#focusableItems[this.#activeIndex - 1]
+        : this.lastItem
     );
   }
 
+  /**
+   * whether filtering is case sensitive
+   */
+  set caseSensitive(caseSensitive: boolean) {
+    if (this.#caseSensitive !== caseSensitive) {
+      this.#caseSensitive = caseSensitive;
+    }
+  }
+
+  get caseSensitive() {
+    return this.#caseSensitive;
+  }
+
+  private static hosts = new WeakMap<ReactiveControllerHost & HTMLElement, RovingTabindexController>();
+
   constructor(public host: ReactiveControllerHost & HTMLElement) {
+    const instance = RovingTabindexController.hosts.get(host);
+    if (instance) {
+      return instance;
+    }
+    RovingTabindexController.hosts.set(host, this);
     this.host.addController(this);
   }
 
+  #nextMatchingItem(key: string) {
+    const items = [...this.#focusableItems];
+    const sequence = [...items.slice(this.#itemIndex - 1), ...items.slice(0, this.#itemIndex - 1)];
+    const regex = new RegExp(`^${key}`, this.#caseSensitive ? '' : 'i');
+    const first = sequence.find(item => {
+      const option = item as HTMLElement;
+      return !option.hasAttribute('disabled') && !option.hidden && option.textContent?.match(regex);
+    });
+    return first || undefined;
```

Comment:
Given that the `find` will return an element or `undefined` already don't think we need this logical OR.

```Suggestion
    return first;
```